libs/Texture/Texture.cpp

```diff
@@ -24,68 +23,128 @@ GLuint Texture::GetTextureId()
 	return m_texture_id;
 }
 
+void Texture::UnBind()
+{
+	glActiveTexture(GL_TEXTURE0);
+	glBindTexture(m_texture_target, 0);
+}
+
 int Texture::Load2DTexture()
 {
-	
 
 	int width = 0;
 	int height = 0;
 	int bbp = 0;
-	//pointer to the image data
-	unsigned char * bits(0);
+	// pointer to the image data
+	unsigned char *bits(0);
 
-	//retrieve the image data
-	//if this somehow one of these failed (they shouldn't), return failure
+	// retrieve the image data
+	// if this somehow one of these failed (they shouldn't), return failure
 	GLuint textureID = LoadTexture(m_file_name, width, height, &bits, bbp);
 	if (textureID == -1)
 	{
 		assert(false && "Image failed to load 2");
 	}
 
-	
 	return textureID;
 }
 
-int Texture::LoadTexture(const std::string & fileName, 
-	int & width, int & height, unsigned char ** data, int& bbp)
+int Texture::Load3DTexture(const std::vector<std::string> &paths)
+{
+	if (paths.size() == 0)
+	{
+		assert(false && "no paths to load from");
+	}
+
+	GLsizei width, height, depth = paths.size();
+
+	std::vector<image> formatedImages(paths.size());
+
+	// load and format each image
+	for (int i = 0; i < paths.size(); ++i)
+	{
+		BYTE *bits(0);
+		int bbp = 0;
+		if (!LoadTexture(paths[i], width, height, &bits, bbp))
+		{
+			assert(false && "Image failed to load 2");
+		}
+		image img(width, height, bits);
+		formatedImages[i] = img;
+	}
+
+	GLuint textureID;
+	glGenTextures(1, &textureID);
+	glBindTexture(GL_TEXTURE_2D_ARRAY, textureID);
+	// Create storage for the texture. (100 layers of 1x1 texels)
+	glTexStorage3D(GL_TEXTURE_2D_ARRAY,
+				   1,			  // No mipmaps as textures are 1x1
+				   GL_RGBA8,	  // Internal format
+				   width, height, // width,height
+				   depth		  // Number of layers
+	);
+
+	for (unsigned int i = 0; i < formatedImages.size(); ++i)
+	{
+		// Specify i-essim image
+		glTexSubImage3D(GL_TEXTURE_2D_ARRAY,
+						0,																// Mipmap number
+						0, 0, i,														// xoffset, yoffset, zoffset
+						formatedImages[i].imageWidth, formatedImages[i].imageHeight, 1, // width, height, depth
+						GL_RGB,															// format
+						GL_UNSIGNED_BYTE,												// type
+						formatedImages[i].imageData);									// pointer to data
```

Comment:
I'd run the formatter through here too. It may be how the formatter is tabbing things but it doesn't seem like the right spacing to me. Either way I'd adjust the tabbing the comments

Reply:
I reformatted the file.